### PR TITLE
tarsnap 1.0.36.1

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

**Comment by DomT4** [comment]
Can you grab the `xz` and `sse` changes from #43160, Thanks.

**Comment by dunn** [comment]
Done!

File: Library/Formula/tarsnap.rb [hunk 1]
```diff
@@ -11,14 +11,32 @@ class Tarsnap < Formula
...
+  depends_on "xz" => :optional
 
   def install
-    system "./configure", "--disable-dependency-tracking",
-                          "--enable-sse2",
-                          "--prefix=#{prefix}",
-                          "--sysconfdir=#{etc}"
+    system "autoreconf", "-iv" if build.head?
+
+    args = %w[
```
Comment (DomT4) [inline Library/Formula/tarsnap.rb]: Need a big `W` rather than a little one here. The little one ignores the variables tucked inside the `#{}` brackets.

Comment (dunn) [inline Library/Formula/tarsnap.rb]: Oh, right.

**Comment by DomT4** [comment]
Thanks! SSE is now scooped up at runtime, which is nicer behaviour, and configure runs around looking for LZMA on the system opportunistically which is less nice behaviour :smile_cat:.

**Comment by dunn** [comment]
Oh, neat. Sorry for ~scooping~ you on the PR 🏂

**Comment by DomT4** [comment]
No worries, It happens :smiley_cat:

**Comment by dunn** [comment]
Not sure why this only happened on Yosemite http://bot.brew.sh/job/Homebrew%20Pull%20Requests/31971/version=yosemite/testReport/junit/brew-test-bot/yosemite/install___tarsnap_1_0_36_1_yosemite_bottle_tar_gz/ cc @woodruffw @mistydemeo ?

**Comment by woodruffw** [comment]
@dunn Hmm. Traceback at `formula_installer.rb:187` suggests that `formula.bottle` is `nil`. From there, it looks like `Formula#bottle` calls `Formula#bottled?` which checks for a tag matching the OS X version in the `bottle` DSL (i.e. `sha1 <hash> => :yosemite`).
Since I don't see a `:yosemite` tag in the formula, my best guess is that you need one in order to pour the bottle.

**Comment by DomT4** [comment]
Just encountered this in #43165. There's a decent chunk of formula that don't have Yosemite bottles already, including every new submission. Is this going to crop up in all of these situations?

**Comment by woodruffw** [comment]
@DomT4 That's what it looks like. Something like this should fix it:

```diff
diff --git a/Library/Homebrew/formula_installer.rb b/Library/Homebrew/formula_installer.rb
index 1198e3f..1b4895c 100644
--- a/Library/Homebrew/formula_installer.rb
+++ b/Library/Homebrew/formula_installer.rb
@@ -184,7 +184,7 @@ class FormulaInstaller
 
     if pour_bottle?(:warn => true)
       begin
-        install_relocation_tools unless formula.bottle.skip_relocation?
+       install_relocation_tools if (formula.bottle && !formula.bottle.skip_relocation?) || !formula.bottle
         pour
       rescue => e
         raise if ARGV.homebrew_developer?
```

That would pull the relocation tools in as a failsafe when `formula.bottle` is `nil`.

**Comment by xu-cheng** [comment]
Bottle problem should be fixed in 5f004b76c1bd6bc28b32d546a5f1406418ab0f19.

**Comment by dunn** [comment]
Thank you all for the help here!

**Comment by woodruffw** [comment]
Thank you @xu-cheng!

**Comment by DomT4** [comment]
Thanks both. I dropped off Github to get some :sleeping: and missed the tail end of this discussion.

**Comment by DomT4** [comment]
Going to break the semi-rule that maintainers don't generally merge each other's PRs here, since `tarsnap` fixed a couple of vulnerabilities in this release. Apologies to Alex!

Merged in https://github.com/Homebrew/homebrew/commit/3c77d914a99e09ba9cc46010f72dcfd74ea64105.

**Comment by dunn** [comment]
Ha, no problem, I fell asleep too.

**Pull request closed without merging**